lib/fs.js

```diff
@@ -811,16 +811,26 @@ function readvSync(fd, buffers, position) {
  *   ) => any} callback
  * @returns {void}
  */
-function write(fd, buffer, offset, length, position, callback) {
+function write(fd, buffer, offsetOrOptions, length, position, callback) {
   function wrapper(err, written) {
     // Retain a reference to buffer so that it can't be GC'ed too soon.
     callback(err, written || 0, buffer);
   }
 
   fd = getValidatedFd(fd);
 
+  let offset = offsetOrOptions;
   if (isArrayBufferView(buffer)) {
     callback = maybeCallback(callback || position || length || offset);
+
+    if (typeof offset === 'object') {
+      ({
+        offset = 0,
+        length = buffer.byteLength - offset,
+        position = null
```

Comment:
nit: trailing commas
```suggestion
        position = null,
```